tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,327 @@
+"""
+Test the auto-restart feature of containers 
+"""
+import time
+import pytest
+import logging
+
+from collections import defaultdict
+
+from common.utilities import wait_until
+from common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_STOP_THRESHOLD_SECS = 30
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+CMD_CONTAINER_FEATURE_AUTORESTART = "show container feature autorestart"
+
+def get_critical_group_and_process_list(duthost, container_name):
+    """
+    @summary: Get critical group and process lists by parsing the 
+              critical_processes file in the specified container
+    @return: Two lists which include the critical process and critical groups respectively
+    """
+    critical_group_list = []
+    critical_process_list = []
+
+    file_content = duthost.shell("docker exec {} bash -c '[ -f /etc/supervisor/critical_processes ] \
+        && cat /etc/supervisor/critical_processes'".format(container_name), module_ignore_errors=True)
+    for line in file_content["stdout_lines"]:
+        line_info = line.strip(' \n').split(':')
+        if len(line_info) != 2:
+            pytest_assert(False, "Syntax of the line {} in critical_processes \
+                file is incorrect.".format(line))
+        
+        identifier_key = line_info[0].strip()
+        identifier_value = line_info[1].strip()
+        if identifier_key == "group" and identifier_value:
+            critical_group_list.append(identifier_value)
+        elif identifier_key == "program" and identifier_value:
+            critical_process_list.append(identifier_value)
+        else:
+            pytest_assert(False, "Syntax of the line {} in critical_processes \
+                file is incorrect.".format(line))
+ 
+    return critical_group_list, critical_process_list
+
+def get_group_program_info(duthost, container_name, critical_group):
+    """
+    @summary: Get critical program names, running status and their pids by analyzing the command
+              output of "docker exec <container_name> supervisorctl"
+    @return: A dictionary where keys are the critical program names and values
+             are their running status and pids
+    """
+    group_program_info = defaultdict(list)
+    program_name = None
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(critical_group) != -1:
+            program_name = program_info.split()[0].split(':')[1].strip(' \n')
```

Comment:
Is there a reason to define the chars to strip as `' \n'`? By default, `strip()` will strip all leading and trailing whitespace.